src/java/org/apache/cassandra/security/SSLFactory.java

```diff
@@ -80,20 +94,26 @@ public static SSLServerSocket getServerSocket(EncryptionOptions options, InetAdd
     }
 
     /** Create a socket and connect */
-    public static SSLSocket getSocket(EncryptionOptions options, InetAddress address, int port, InetAddress localAddress, int localPort) throws IOException
+    public static SSLSocket getSocket(EncryptionOptions options, InetAddress address, int port, InetAddress localAddress, int localPort) throws Exception
     {
         SSLContext ctx = createSSLContext(options, true);
-        SSLSocket socket = (SSLSocket) ctx.getSocketFactory().createSocket(address, port, localAddress, localPort);
-        prepareSocket(socket, options, address);
+        InetAddress mappedAddress = CrossVpcIpMappingHandshaker.instance.maybeSwapAddress(address);
+        SSLSocket socket;
```

Comment:
why is socket defined separately from its type declaration? (and in below functions)

Reply:
Oversight from previous commit